packages/common-ts/src/base-service/base-service-v2.ts

```diff
@@ -389,12 +393,36 @@ export abstract class BaseServiceV2<
 
       // Metrics.
       // Will expose a /metrics endpoint by default.
+
+      const router = express.Router()
+
+      if (this.routes) {
+        this.routes(router)
+      }
+
       app.use(
         promBundle({
           promRegistry: this.metricsRegistry,
           includeMethod: true,
           includePath: true,
           includeStatusCode: true,
+          bypass: (req) => {
+            // Note we are bypassing all routes that aren't at /api/* by only including the router.stack
+            for (const r of router.stack) {
+              if (req.path.includes(r.path)) {
+                return false
+              }
+            }
+            return true
+          },
```

Comment:
Does this mean we're not recording metrics for paths that don't exist?

Reply:
Yes this was a specific request from @ben-chain because people be pinging our server for routes that don't exist all the time.